### ISSUE TITLE
Standalone program fails with ModuleNotFoundError: 'sklearn.neighbors._quad_tree'

### ISSUE THREAD

**Issue by jkissi** [Nuitka/Nuitka#1813]
Hello!
I am attempting to compile a fairly substantial program using nuitka. This process seems to complete successfully, with no major warnings and very quickly (usually within 20 mins). Upon completing compilation and running the executable in the .dist/ folder using a git bash terminal, I receive the following error:

```
$ ./nnse.exe --help
Traceback (most recent call last):
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\nnse.py", line 18, in <module>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\graphical\interface.py", line 26, in <module graphical.interface>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\pipeline\experimental_framework.py", line 13, in <module pipeline.experimental_framework>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\pipeline\pipeline.py", line 22, in <module pipeline.pipeline>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\pipeline\functional.py", line 14, in <module pipeline.functional>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\hdbscan\__init__.py", line 1, in <module hdbscan>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\hdbscan\hdbscan_.py", line 37, in <module hdbscan.hdbscan_>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\hdbscan\plots.py", line 9, in <module hdbscan.plots>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\sklearn\manifold\__init__.py", line 9, in <module sklearn.manifold>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_3\nnse.dist\sklearn\manifold\_t_sne.py", line 30, in <module sklearn.manifold._t_sne>    
  File "sklearn\manifold\_barnes_hut_tsne.pyx", line 1, in init sklearn.manifold._barnes_hut_tsne
ModuleNotFoundError: No module named 'sklearn.neighbors._quad_tree'
```
It doesn't seem to matter what command I give my program, the output is the same. The output is confusing because I can see that the sklearn module is referenced as compilation is occurring, so I don't understand how this module is not included. Is there some way I can manually include it? Further details are below...

* Nuitka version, full Python version, flavor, OS, etc. as output by *this*
  command.
```
1.1
Commercial: None
Python: 3.10.4 | packaged by conda-forge | (main, Mar 30 2022, 08:38:02) [MSC v.1916 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: D:\installs\programs\Anaconda3\envs\nnse\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
```

* How did you install Nuitka and Python

**All related libraries (vedo, nuitka, python) are installed using Anaconda command line in an anaconda environment**

* The specific PyPI names and versions

  It should be taken from this output if there specific packages involved, e.g.
  `numpy`, you are expected to shorten this to the relevant ones.

  > python -m pip freeze

Wasn't sure what to remove, so left the entire output here.
```
certifi==2022.9.24
click @ file:///C:/ci/click_1646056762388/work
colorama @ file:///home/conda/feedstock_root/build_artifacts/colorama_1655412516417/work    
cycler @ file:///home/conda/feedstock_root/build_artifacts/cycler_1635519461629/work        
dearpygui==1.6.2
Deprecated @ file:///home/conda/feedstock_root/build_artifacts/deprecated_1632758190537/work
fonttools==4.25.0
hdbscan @ file:///D:/bld/hdbscan_1644613083537/work
joblib @ file:///tmp/build/80754af9/joblib_1635411271373/work
kiwisolver @ file:///D:/bld/kiwisolver_1657953239132/work
laspy @ file:///home/conda/feedstock_root/build_artifacts/laspy_1656594965604/work
loguru @ file:///D:/bld/loguru_1649442989339/work
matplotlib @ file:///D:/bld/matplotlib-suite_1659031634610/work
mkl-fft==1.3.1
mkl-random @ file:///C:/ci_310/mkl_random_1643050563308/work
mkl-service==2.4.0
munkres==1.1.4
Nuitka @ file:///D:/bld/nuitka_1664172646218/work
numpy @ file:///C:/ci/numpy_and_numpy_base_1652784037925/work
opencv-python==4.5.5
ordered-set @ file:///home/conda/feedstock_root/build_artifacts/ordered-set_1643221357603/work
packaging @ file:///home/conda/feedstock_root/build_artifacts/packaging_1637239678211/work
pandas @ file:///D:/bld/pandas_1656001224800/work
Pillow==9.2.0
Pympler @ file:///home/conda/feedstock_root/build_artifacts/pympler_1658620692385/work
pyntcloud @ file:///home/conda/feedstock_root/build_artifacts/pyntcloud_1659284224615/work
pyparsing @ file:///home/conda/feedstock_root/build_artifacts/pyparsing_1652235407899/work
PyQt5==5.12.3
PyQt5_sip==4.19.18
PyQtChart==5.12
PyQtWebEngine==5.12.1
python-dateutil @ file:///home/conda/feedstock_root/build_artifacts/python-dateutil_1626286286081/work
pytz @ file:///home/conda/feedstock_root/build_artifacts/pytz_1660580468234/work
scikit-learn @ file:///C:/ci_310/scikit-learn_1642249118560/work
scipy @ file:///C:/ci_310/scipy_1642247684282/work
seaborn @ file:///tmp/build/80754af9/seaborn_1629307859561/work
sip==4.19.13
six @ file:///tmp/build/80754af9/six_1644875935023/work
tabulate @ file:///home/conda/feedstock_root/build_artifacts/tabulate_1655900837660/work
threadpoolctl @ file:///Users/ktietz/demo/mc3/conda-bld/threadpoolctl_1629802263681/work
torch==1.12.1
tornado @ file:///D:/bld/tornado_1656937966227/work
typing_extensions @ file:///C:/Windows/TEMP/abs_dd2d0moa85/croots/recipe/typing_extensions_1659638831135/work
vedo @ file:///home/conda/feedstock_root/build_artifacts/vedo_1661755873429/work
vtk===-PKG-VERSION
win32-setctime @ file:///home/conda/feedstock_root/build_artifacts/win32_setctime_1642883564726/work
wincertstore==0.2
wrapt @ file:///D:/bld/wrapt_1651495425905/work
```

* Provide in your issue the Nuitka options used

`python -m nuitka --standalone nnse.py --enable-plugin=numpy --enable-plugin=pyqt5 --enable-plugin=matplotlib  --enable-plugin=anti-bloat --output-dir=/d/life/uwo/ece/phd/code/nuitka/standalone/attempt_3
`

Here is the full compilation output (looks pretty good, to be honest):
```
Nuitka-Options:INFO: Used command line options: --standalone nnse.py --enable-plugin=numpy --enable-plugin=pyqt5 --enable-plugin=matplotlib --enable-plugin=anti-bloat --output-dir=D:/life/uwo/ece/phd/code/nuitka/standalone/attempt_3
Nuitka-Plugins:WARNING: anti-bloat: Plugin is defined as always enabled.
Nuitka-Plugins:WARNING: matplotlib: Plugin is defined as always enabled.
Nuitka:INFO: Starting Python compilation with Nuitka '1.1' on Python '3.10' commercial grade 'not installed'.
Nuitka-Plugins:INFO: multiprocessing: Injecting pre-module load code for module 'multiprocessing':
Nuitka-Plugins:INFO: multiprocessing:     Monkey patching "multiprocessing" load environment.
Nuitka-Plugins:INFO: multiprocessing: Injecting post-module load code for module 'multiprocessing':
Nuitka-Plugins:INFO: multiprocessing:     Monkey patching "multiprocessing" for compiled methods.
Nuitka-Plugins:INFO: pyqt5: Injecting pre-module load code for module 'PyQt5':                   
Nuitka-Plugins:INFO: pyqt5:     Adding binary folder to runtime 'PATH' environment variable for proper Qt loading.
Nuitka-Plugins:INFO: pyqt5: Injecting post-module load code for module 'PyQt5.QtCore':
Nuitka-Plugins:INFO: pyqt5:     Setting Qt library path to distribution folder. We need to avoid loading target
Nuitka-Plugins:INFO: pyqt5:     system Qt plugins, which may be from another Qt version.
Nuitka-Plugins:INFO: pkg-resources: Injecting post-module load code for module 'pkg_resources':                                
Nuitka-Plugins:INFO: pkg-resources:     Registering Nuitka loader with "pkg_resources".
Nuitka:INFO: Completed Python level compilation and optimization.                                                              
Nuitka:INFO: Generating source code for C backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.                                                                 
Nuitka:INFO: Running C compilation via Scons.
Nuitka-Scons:INFO: Backend C compiler: cl (cl 14.3).
Nuitka-Scons:INFO: Backend linking program with 1774 files (no progress information available).
Nuitka-Scons:INFO: Running 'link @C:\\Users\\Jon\\AppData\\Local\\Temp\\tmp31q_dh9p.lnk' took 468.79 seconds
Nuitka-Scons:INFO: Compiled 1774 C files using clcache with 1774 cache hits and 0 cache misses.
Nuitka-Plugins:INFO: data-files: Included data file 'certifi\cacert.pem' due to package data for 'certifi'.
Nuitka-Plugins:INFO: matplotlib: Included 183 data files due to package data for 'matplotlib.
Nuitka-Plugins:INFO: matplotlib: Included data file 'matplotlib\mpl-data\matplotlibrc' due to updated matplotlib config file with backend to use.   
Nuitka-Plugins:INFO: data-files: Included data file 'pandas\io\formats\templates\html.tpl' due to package data directory 'templates' for 'pandas.io.formats'.
Nuitka-Plugins:INFO: data-files: Included data file 'pandas\io\formats\templates\html_style.tpl' due to package data directory 'templates' for 'pandas.io.formats'.
Nuitka-Plugins:INFO: data-files: Included data file 'pandas\io\formats\templates\html_table.tpl' due to package data directory 'templates' for 'pandas.io.formats'.
Nuitka-Plugins:INFO: data-files: Included data file 'pandas\io\formats\templates\latex.tpl' due to package data directory 'templates' for 'pandas.io.formats'.
Nuitka-Plugins:INFO: data-files: Included data file 'pandas\io\formats\templates\latex_longtable.tpl' due to package data directory 'templates' for 
'pandas.io.formats'.
Nuitka-Plugins:INFO: data-files: Included data file 'pandas\io\formats\templates\latex_table.tpl' due to package data directory 'templates' for 'pandas.io.formats'.
Nuitka-Plugins:INFO: data-files: Included 601 data files due to package data directory 'zoneinfo' for 'pytz'.
Nuitka-Plugins:INFO: data-files: Included data file 'torch\bin\asmjit.dll' due to package data directory 'bin' for 'torch'.
Nuitka-Plugins:INFO: data-files: Included data file 'torch\bin\fbgemm.dll' due to package data directory 'bin' for 'torch'.
Nuitka-Plugins:INFO: data-files: Included data file 'torch\bin\protoc.exe' due to package data directory 'bin' for 'torch'.
Nuitka-Plugins:INFO: data-files: Included 7688 data files due to package data directory 'include' for 'torch'.
Nuitka-Plugins:INFO: pyqt5: Including Qt plugins 'iconengines,imageformats,mediaservice,platforms,platformthemes,printsupport,styles' below 'PyQt5\qt-plugins'.
Nuitka-Plugins:INFO: numpy: Found 42 files DLLs from numpy installation.
Nuitka-Plugins:INFO: dll-files: Found 22 files DLLs from torch installation.
Nuitka:INFO: Keeping build directory 'D:\\life\\uwo\\ece\\phd\\code\\nuitka\\standalone\\attempt_3\\nnse.build'.
Nuitka:INFO: Successfully created 'D:\\life\\uwo\\ece\\phd\\code\\nuitka\\standalone\\attempt_3\\nnse.dist\\nnse.exe'.
```

Hopefully, once again, you will be able to help me fix this and compile my program.
Thanks in advance,
Cheers
j

**Comment by kayhayen** [comment]
Can youj provide a reproducer, I am tempted to say that `import sklearn.manifold` fails on Anaconda (and probably only there), but I usually insist that users produce it.

**Comment by jkissi** [comment]
This is the reproduction code (in file **_reproducer.py_**):

```
# This is a test file 
from numpy.random import RandomState
import matplotlib.pyplot as plt
from matplotlib import ticker

from sklearn import manifold, datasets



def run():
    """Test"""
    rng = RandomState(0)
    n_neighbors = 12  # neighborhood which is used to recover the locally linear structure
    n_components = 2  # number of coordinates for the manifold
    n_samples = 1500
    S_points, S_color = datasets.make_s_curve(n_samples, random_state=rng)


    def add_2d_scatter(ax, points, points_color, title=None):
        x, y = points.T
        ax.scatter(x, y, c=points_color, s=50, alpha=0.8)
        ax.set_title(title)
        ax.xaxis.set_major_formatter(ticker.NullFormatter())
        ax.yaxis.set_major_formatter(ticker.NullFormatter())


    def plot_2d(points, points_color, title):
        fig, ax = plt.subplots(figsize=(3, 3), facecolor="white", constrained_layout=True)
        fig.suptitle(title, size=16)
        add_2d_scatter(ax, points, points_color)
        plt.show()

    t_sne = manifold.TSNE(
        n_components=n_components,
        learning_rate="auto",
        perplexity=30,
        n_iter=250,
        init="random",
        random_state=rng,
    )
    S_t_sne = t_sne.fit_transform(S_points)

    plot_2d(S_t_sne, S_color, "T-distributed Stochastic  \n Neighbor Embedding")

if __name__ == '__main__':
    print("This is before main")
    run()
    print("This is after main")
```

When run using:
`python reproducer.py `
the output is the following: 

![nuitka_sklearn_manifold](https://user-images.githubusercontent.com/8246894/192843437-32dd1b0b-a88b-4ac8-8e5e-52eda3d5d90f.PNG)


When compiled using:
`$ python -m nuitka --standalone reproducer.py --enable-plugin=numpy --enable-plugin=pyqt5 --output-dir=/d/life/uwo/ece/phd/code/nuitka/standalone/attempt_4`

And then run from inside the .dist folder, the output is the following:
```
$ ./reproducer.exe
Traceback (most recent call last):
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_4\reproducer.dist\reproducer.py", line 6, in <module>
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_4\reproducer.dist\sklearn\manifold\__init__.py", line 9, in <module sklearn.manifold>    
  File "D:\life\uwo\ece\phd\code\nuitka\standalone\attempt_4\reproducer.dist\sklearn\manifold\_t_sne.py", line 30, in <module sklearn.manifold._t_sne>
  File "sklearn\manifold\_barnes_hut_tsne.pyx", line 1, in init sklearn.manifold._barnes_hut_tsne
ModuleNotFoundError: No module named 'sklearn.neighbors._quad_tree'
```

 Is this an ok reproduction or did I do something wrong here? 

Cheers 
j

**Comment by jkissi** [comment]
Hi Kay, 
I just want to make sure this reproducer I gave was appropriate or whether you need something more from me. Please let me know. 
Cheers 
j

**Comment by kayhayen** [comment]
@jkissi yes it is, the thing with non-immediate reproducers is that I tend to overlook them.

**Comment by kayhayen** [comment]
This was part of the 1.1.5 hotfix release, but not yet reported.

**Comment by jkissi** [comment]
That's great news! Thanks for your efforts here, very much appreciated!!!
